src/classes/Snake.js

```diff
@@ -0,0 +1,80 @@
+/* eslint-disable no-plusplus */
+/* eslint-disable import/extensions */
+/* eslint-disable import/prefer-default-export */
+import { randomGridPosition } from '../../utils/gridPositions.js';
+import { GRID_SIZE } from '../../utils/constants.js';
+
+export class Snake {
+  constructor() {
+    this.snakeBody = [randomGridPosition()];
+    this.newSegments = 0;
+    this.points = 0;
+  }
+
+  addSegments() {
+    for (let i = 0; i < this.newSegments; i++) {
+      this.snakeBody.push({ ...this.snakeBody[this.snakeBody.length - 1] });
+    }
+
+    this.newSegments = 0;
+  }
+
+  movement(inputDirection) {
+    this.addSegments();
+
+    for (let i = this.snakeBody.length - 2; i >= 0; i--) {
+      this.snakeBody[i + 1] = { ...this.snakeBody[i] };
+    }
+
+    this.snakeBody[0].x += inputDirection.x;
+    this.snakeBody[0].y += inputDirection.y;
+
+    // wall mirroring
+    if (this.snakeBody[0].x < 1) {
+      this.snakeBody[0].x = GRID_SIZE;
+    } else if (this.snakeBody[0].x > GRID_SIZE) {
+      this.snakeBody[0].x = 1;
+    }
+
+    if (this.snakeBody[0].y < 1) {
+      this.snakeBody[0].y = GRID_SIZE;
+    } else if (this.snakeBody[0].y > GRID_SIZE) {
+      this.snakeBody[0].y = 1;
+    }
+  }
+
+  render(gameBoard) {
+    this.snakeBody.forEach((segment) => {
+      const snakeElement = document.createElement('div');
+      snakeElement.style.gridRowStart = segment.y;
+      snakeElement.style.gridColumnStart = segment.x;
+      snakeElement.classList.add('snake');
+      gameBoard.appendChild(snakeElement);
```

Comment:
imagine that snake growing more and more and what happened here we do some performance cost operation that intersecting the dom, so  to optimize this you can append new nodes to the node with `append` method at once.


```
const segmentsToElements = this.snakeBody.reduce((acc, segment) => {
      const snakeElement = document.createElement("div");
      snakeElement.style.gridRowStart = segment.y;
      snakeElement.style.gridColumnStart = segment.x;
      snakeElement.classList.add("snake");

      return [...acc, snakeElement];
    }, []);

    gameBoard.append(...segmentsToElements);
```